### PR TITLE
Move the full contribution guide for new 3rd party app support to GitHub.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

File: dev-docs/new-receivers.md [hunk 1]
```diff
@@ -1,4 +1,200 @@
...
+    pipelines:
+      apache:
+        receivers:
+        - apache
+```
+
+#### 2.3 Validations against invalid configurations for the receiver(s)
+
+Sample validation cases are like:
+- A required parameter is not set for the given receiver (s)
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:121]:
```suggestion
- A required parameter is not set for the given receiver(s)
```

File: dev-docs/new-receivers.md [hunk 2]
```diff
@@ -1,4 +1,200 @@
-# Receiver How-To
+# New 3rd Party Application Support How-To
+
+## Steps
+
+### 1. Research what telemetry the application generates
+
+Figure out the following details for the application to facilitate the config design later.
+
+- How many versions of the application are dominating in the industry, and whether we need to support more than one versions.
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:9]:
```suggestion
- How many versions of the application are dominating in the industry, and whether we need to support more than one version.
```

File: dev-docs/new-receivers.md [hunk 3]
```diff
@@ -1,4 +1,200 @@
...
+    - Log file paths: The common log file paths for these logs.
+    - Log formats: The log formats inside these files, and available customization options if users want to change the
+      log format.
+  - Metrics
+    - What metrics: Once installed, what metrics the application exposes by default. Any additional application settings
+      users can change to expose additional metrics.
+    - How to expose metrics. If the application does not expose metrics by default (e.g. Apache does not), what
+      application settings users need to specify to enable it.
+    - Required user configurations (e.g. username) for the Ops Agent to talk to the application to get the metrics
+      (e.g. username, password, database name, port etc.)
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:23]:
```suggestion
      (e.g. username, password, database name, port, etc.)
```

File: dev-docs/new-receivers.md [hunk 4]
```diff
@@ -1,4 +1,200 @@
...
+    apache_error:
+      type: apache_error
+      included_paths: /var/log/my_error_log # Optional.
+      format: # Optional
+  service:
+    pipelines:
+      apache:
+        receivers:
+        - apache_access
+        - apache_error
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:103]:
```suggestion
        receivers: [apache_access, apache_error]
```

File: dev-docs/new-receivers.md [hunk 5]
```diff
@@ -1,4 +1,200 @@
...
+    apache:
+      type: apache
+      endpoint: #  Optional
+      username: # Optional
+      password: # Optional
+  service:
+    pipelines:
+      apache:
+        receivers:
+        - apache
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:115]:
```suggestion
        receivers: [apache]
```

File: dev-docs/new-receivers.md [hunk 6]
```diff
@@ -1,4 +1,200 @@
...
+```
+
+#### 2.3 Validations against invalid configurations for the receiver(s)
+
+Sample validation cases are like:
+- A required parameter is not set for the given receiver (s)
+- An unknown parameter is set for the receiver (s)
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:122]:
```suggestion
- An unknown parameter is set for the receiver(s)
```

File: dev-docs/new-receivers.md [hunk 7]
```diff
@@ -1,4 +1,200 @@
...
+For each application, the documentation needs to cover the following:
+- How to enable logs / metrics ingestion for this application
+  - If the application does not expose metrics by default, what users need to do to the application to enable it. This
+    might be no-op for many applications.
+  - How users customize the Ops Agent config file to enable scraping and ingesting telemetry for this application. This
+    contains details about the new receiver type(s). For example:
+  - The default log file paths the logging receiver(s) expect logs to be at
+  - The default log formats the logging receiver(s) expect the logs to be in
+  - The default metrics endpoint / port the metrics receiver expect metrics to be exposed at
+  - Available parameters of the receiver(s) to control additional behaviors
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:185]:
```suggestion
    - The default log file paths the logging receiver(s) expect logs to be at
    - The default log formats the logging receiver(s) expect the logs to be in
    - The default metrics endpoint / port the metrics receiver expect metrics to be exposed at
    - Available parameters of the receiver(s) to control additional behaviors
```

File: dev-docs/new-receivers.md [hunk 8]
```diff
@@ -1,4 +1,200 @@
...
+  - Sample pipeline and receiver configuration snippets
+- How to query logs and metrics for this application once ingested
+  - When it comes to querying the logs and metrics data points, which log name, log labels, metrics namespace, metrics
+    names, metrics labels to filter by.
+  - Which dashboards contain these metrics by default.
+- Common failure cases specific to this application and how to troubleshoot them
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:191]:
```suggestion
- Common failure cases specific to this application and how to troubleshoot them.
```

File: dev-docs/new-receivers.md [hunk 9]
```diff
@@ -1,4 +1,200 @@
...
+  For each of the failure cases, document how users could detect that failure condition (e.g. via errors in the Ops
+  Agent log, via Ops Agent’s own health metrics), and how to fix it. Sample failure condition includes:
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:193]:
```suggestion
  Agent log, via Ops Agent’s own health metrics), and how to fix it. Sample failure conditions include:
```

File: dev-docs/new-receivers.md [hunk 10]
```diff
@@ -10,36 +206,31 @@ Generic code is shared across applications:
...
- 
+
 import "github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
- 
+
 type MetricsReceiverIis struct {
 	ConfigComponent `yaml:",inline"`
- 
+
 	MetricsReceiverShared `yaml:",inline"`
```
Comment (igorpeshansky) [inline dev-docs/new-receivers.md:227]:
```suggestion
	// This is a convenience struct with common fields like CollectionInterval.
	MetricsReceiverShared `yaml:",inline"`
```

File: dev-docs/new-receivers.md [hunk 11]
```diff
@@ -10,36 +206,31 @@ Generic code is shared across applications:
...
 }
- 
+
 func (r MetricsReceiverIis) Type() string {
 	return "iis"
```
Comment (igorpeshansky) [inline dev-docs/new-receivers.md:231]:
```suggestion
	// This is the string that will identify this receiver in user configs.
	return "iis"
```

File: dev-docs/new-receivers.md [hunk 12]
```diff
@@ -10,36 +206,31 @@ Generic code is shared across applications:
...
+And then one file per 3rd party application integration should contain the config struct(s) for that integration and any
+integration-specific business logic. These files are defined inside https://github.com/GoogleCloudPlatform/ops-agent/tree/master/confgenerator
 
 - One file per application integration, defining receiver and processor struct(s) as needed for that application including both logging and metrics
 - Validation is performed with struct tags on each config struct (~zero per-type validation code)
 - Config structs have a method to generate OT and fluentbit config from the config struct. See details in the Pipelines() method below
-- Each application register its own receiver by specifying the receiver in the init() function of its own file. Adding support for a new application only needs to touch this one new file, unless it needs to add common utils. 
+- Each application register its own receiver by specifying the receiver in the init() function of its own file. Adding support for a new application only needs to touch this one new file, unless it needs to add common utils.
```
Comment (davidbtucker) [inline dev-docs/new-receivers.md:215]:
```suggestion
- Each application registers its own receiver by specifying the receiver in the init() function of its own file. Adding support for a new application only needs to touch this one new file, unless it needs to add common utils.
```

**Review by igorpeshansky — COMMENTED** [review]
A couple of suggested comments to guide the implementors.

**Review by davidbtucker — APPROVED** [review]
Just fixed some typos.

**Pull request merged**